src/components/menu/menu-initial-focus/menu-initial-focus.directive.ts

```diff
@@ -0,0 +1,29 @@
+import { Directive, ElementRef, OnInit, Renderer2 } from '@angular/core';
+import { MenuComponent } from '../menu/menu.component';
+
+@Directive({ selector: '[uxMenuInitialFocus]' })
+export class MenuInitialFocusDirective implements OnInit {
+    constructor(
+        private readonly _menu: MenuComponent,
+        private readonly _elementRef: ElementRef<HTMLElement>,
+        private readonly _renderer: Renderer2
+    ) {}
+
+    ngOnInit(): void {
+        this.ensureFocusable();
+
+        // Focus the host element when the parent menu is opened.
+        this._menu.opened.pipe().subscribe(() => {
```

Comment:
Think we need to unsubscribe from this observable 😊

Reply:
I can make this change, thanks Ashley